app/models/setting.rb

```diff
@@ -201,6 +201,15 @@ def self.find_or_default(name)
     end
   end
 
+  # clears all settings from the cache. #delete_matched might not be supported
+  # by every implementation. in this case we fall back to clearing the entire
+  # cache. if #clear is also not supported we just yawn and fail together.
+  def self.clear_cache!
+    Rails.cache.delete_matched /\/openproject\/settings\/.*/
+  rescue NotImplementedError
```

Comment:
`rescue NotImplementedError, NoMethodError`

we also need to rescue from `NoMethodError` because implementations not necessarily descend from the base class.